src/libraries/System.Net.Http/src/System/Net/Http/BrowserHttpHandler/BrowserHttpHandler.cs

```diff
@@ -0,0 +1,594 @@
+            set => throw new PlatformNotSupportedException("Property MaxResponseDrainSize is not supported.");
+        }
+
+        public TimeSpan ResponseDrainTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property ResponseDrainTimeout is not supported.");
+            set => throw new PlatformNotSupportedException("Property ResponseDrainTimeout is not supported.");
+        }
+
+        public int MaxResponseHeadersLength
+        {
+            get => throw new PlatformNotSupportedException("Property MaxResponseHeadersLength is not supported.");
+            set => throw new PlatformNotSupportedException("Property MaxResponseHeadersLength is not supported.");
+        }
+
+        public SslClientAuthenticationOptions SslOptions
+        {
+            get => throw new PlatformNotSupportedException("Property SslOptions is not supported.");
+            set => throw new PlatformNotSupportedException("Property SslOptions is not supported.");
+        }
+
+        public TimeSpan PooledConnectionLifetime
+        {
+            get => throw new PlatformNotSupportedException("Property PooledConnectionLifetime is not supported.");
+            set => throw new PlatformNotSupportedException("Property PooledConnectionLifetime is not supported.");
+        }
+
+        public TimeSpan PooledConnectionIdleTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property PooledConnectionLifetime is not supported.");
+            set => throw new PlatformNotSupportedException("Property PooledConnectionLifetime is not supported.");
+        }
+
+        public TimeSpan ConnectTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property ConnectTimeout is not supported.");
+            set => throw new PlatformNotSupportedException("Property ConnectTimeout is not supported.");
+        }
+
+        public TimeSpan Expect100ContinueTimeout
+        {
+            get => throw new PlatformNotSupportedException("Property Expect100ContinueTimeout is not supported.");
+            set => throw new PlatformNotSupportedException("Property Expect100ContinueTimeout is not supported.");
+        }
+
+        public IDictionary<string, object?> Properties => throw new PlatformNotSupportedException("Property Properties is not supported.");
+        protected internal override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            return DoFetch(request, cancellationToken);
+        }
+
+        private async Task<HttpResponseMessage> DoFetch(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+
+            var tcs = new TaskCompletionSource<HttpResponseMessage>();
+
+            // Wrap the cancellationToken in a using so that it can be disposed of whether
+            // we successfully fetched or failed trying.
+            using (cancellationToken.Register(() => tcs.TrySetCanceled()))
```

Comment:
Is there no way to cancel the actual operation via the browser?  This approach is going to cancel the task we return to the caller, but the underlying operation may still be on-going.

Reply:
(If there is a way, it's likely we can also remove the tcs.)

Reply:
Yes there is a way to cancel from the browser side.  

```
                    JSObject abortController = new HostObject("AbortController");
                    JSObject signal = (JSObject)abortController.GetObjectProperty("signal");
                    requestObject.SetObjectProperty("signal", signal);
                    signal.Dispose();

                    CancellationTokenSource abortCts = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
                    CancellationTokenRegistration abortRegistration = abortCts.Token.Register((Action)(() =>
                    {
                        if (abortController.JSHandle != -1)
                        {
                            abortController.Invoke("abort");
                            abortController?.Dispose();
                        }
                        wasmHttpReadStream?.Dispose();
                    }));
```

But do not see a way to get rid of tcs right now.  Will need to look further and look into a recent change to this for gRPC and streaming.

Reply:
> But do not see a way to get rid of tcs right now. 

Why is the tcs necessary?

Reply:
Will need to look further and look into a recent change to this for gRPC and streaming.

Reply:
Removed the tcs as referenced above.  Was not needed as you pointed out.

Addressed in recent commit